Added quantization for evaluation script

rohansjoshi opened the pull request:
Summary:
Added quantization to evaluation script. Quantization causes deterioriation in accuracy

On wikitext task:
| Model Name | max_seq_len | ptq | word_perplexity
|----------|----------|----------|-----------|
| Llama 3.2-1B Instruct  | 128   | 16a4w |  5821003.055178451 |
| Llama 3.2-1B Instruct  | 128   | 16a4w_block |  5396240.078572427 |
| Llama 3.2-1B Instruct  | 128   | 8a8w |  533154.970440251 |

Differential Revision: D76837572

cccclai reviewed (approved):
Thank you for the ppl summary table

rohansjoshi committed:
Added quantization for evaluation script (#11822)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/11822

Added quantization to evaluation script. Quantization causes deterioriation in accuracy

On wikitext task:
| Model Name | max_seq_len | ptq | word_perplexity
|----------|----------|----------|-----------|
| Llama 3.2-1B Instruct  | 128   | 16a4w |  5821003.055178451 |
| Llama 3.2-1B Instruct  | 128   | 16a4w_block |  5396240.078572427 |
| Llama 3.2-1B Instruct  | 128   | 8a8w |  533154.970440251 |

Reviewed By: cccclai

Differential Revision: D76837572
